include/EASTL/vector.h

```diff
@@ -890,7 +890,19 @@ namespace eastl
 			else if(n < (size_type)(mpEnd - mpBegin))
 				resize(n);
 
+		#if EASTL_MOVE_SEMANTICS_ENABLED
+			this_type     temp;
+			size_type     nNewSize = size();
+			pointer const pNewData = DoAllocate(nNewSize);
+
+			pointer pNewEnd = eastl::uninitialized_move_ptr(mpBegin, mpEnd, pNewData);
+
+			temp.mpBegin    = pNewData;
+			temp.mpEnd      = pNewEnd;
+			temp.mpCapacity = pNewEnd;
+		#else
 			this_type temp(*this);  // This is the simplest way to accomplish this, 
```

Comment:
Indeed shrink_to_fit() function seems to do the job. And the exact same job as it's needed here. So I remove the part of code which create a temp vector and swap it and just call shrink_to_fit().
Tell me if it's all ok for you.